mpl_typst/backend.py

```diff
@@ -126,9 +142,51 @@ def __exit__(self, exc_type, exc_value, traceback):
                     height=Scalar(self.height, 'in'))
         expr.to_string(self.writer)
 
+    def draw_gouraud_triangles(self, gc, triangles_array, colors_array,
+                               transform):
+        pass
+
     def draw_image(self, gc: GraphicsContextBase, x: float, y: float,
                    im: ArrayLike, transform: Affine2DBase | None = None):
-        raise NotImplementedError
+        # This methods closely follows the SVG backend
+
+        h, w = im.shape[:2]
+
+        if w == 0 or h == 0:
+            return
+
+        # TODO Clips as soon as Typst supports them
+        # TODO Links, Transforms (except translate)
+
+        data = ''
+
+        img = ImageOps.flip(Image.fromarray(im))
```

Comment:
why do you flip the image here?

Reply:
It is due to axis direction. Another option is to invert axis and keep image as is.